### ISSUE TITLE
Pod labels are not updated correctly after failover

### ISSUE THREAD

**Issue by GeorgFleig** [CrunchyData/postgres-operator#1218]
**Describe the bug**
When a failover happened, the labels of the pod are not updated correctly. This affects the `role` and the `service-name` labels. When the labels are wrong, traffic to the service is send to the wrong pods.

I've observed two cases so far. The first one happens every time, the second one was only observed once so far.

### Case 1: `service-name` label of old primary is not updated

Fresh deployment with two replicas (just one shown in this example).

Before the failover:

```
primary pod: pg1-5cd4455dbf-pxqj5
    role: master
    service-name: pg1

a replica pod: pg1-ufwt-65f9dff5f6-nl47f
    role: replica
    service-name: pg1-replica
```

Triggered failover with `pgo failover`. Failover looks good when looking at the postgres/patroni logs. However the labels are messed up afterwards:

```
new replica (old primary): pg1-5cd4455dbf-pxqj5
    role: replica
    service-name: pg1

new primary (old replica): pg1-ufwt-65f9dff5f6-nl47f
    role: master
    service-name: pg1
```

The `service-name` label of the old primary was not updated. Client traffic is now routed to both pods. This can be reproduced consistently.

### Case 2: `role` and `service-name` labels are mixed up

In another case I caused a failover by rebooting the k8s node that was running the master pod. This pg2 cluster has synchronous replication enabled.

Before the failover:

```
primary pod: pg2-5448766fbf-nl4ns
    role: master
    service-name: pg2

a replica pod: pg2-khsa-7684bfd89b-kchwn
    role: replica
    service-name: pg2-replica
```

After the failover:

```
new replica (old primary): pg2-5448766fbf-nl4ns
    role: replica
    service-name: pg2

new primary (old replica): pg2-khsa-7684bfd89b-kchwn
    role: master
    service-name: pg2-replica
```

This could only be observed once so far.

**To Reproduce**
1. Create a deployment with at least one replica
2. Trigger a failover
3. Check labels of the two postgres pods

**Expected behavior**
Labels should be set correctly so client traffic is routed to the primary pod only.

**Please tell us about your environment:**

* Operating System: Ubuntu 18.04
* Where is this running: Private cloud
* Storage being used: Ceph
* Container Image Tag: 4.2.0
* PostgreSQL Version: 11.6
* Platform (Docker, Kubernetes, OpenShift): k8s 1.15

**Comment by andrewlecuyer** [comment]
@GeorgFleig, as of version 4.2 of the PostgreSQL Operator, the `service-name` label is no longer utilized by the primary and replica services, and we therefore plan to remove this label in a future release.  If you look at the selector for either the primary and replica service, you will see that the `service-name` label is not utilized.  For example, for a cluster named `mycluster1`, the associated primary service will have a selector such as the following:

```yaml
selector:
    pg-cluster: mycluster1
    role: master
```

...and the replica service will have a selector as follows:

```yaml
selector:
    pg-cluster: mycluster1
    role: replica
```

Therefore, with the selectors selectors currently utilized by the primary and replica services as shown above, the service will route traffic to the proper primary or replica pods.

**Comment by GeorgFleig** [comment]
@andrewlecuyer the way you describe the selector of the service makes total sense and this is also what I modified my services to to fix the issues.

However it is not the state the services are created in 4.2.0. I just verified it again with a new test cluster. The selector is just based on `service-name` and therefore traffic is routed to the wrong pods.

Digging a little into the pgo config files I found the underlying issue. My PGO was deployed using ansible. The ansible deployment uses different template files for the k8s resources than the bash PGO deployment.

The two files got out of sync:
https://github.com/CrunchyData/postgres-operator/blob/master/conf/postgres-operator/cluster-service.json
https://github.com/CrunchyData/postgres-operator/blob/master/ansible/roles/pgo-operator/files/pgo-configs/cluster-service.json
